python_modules/dagster/dagster/_grpc/types.py

```diff
@@ -732,3 +734,151 @@ def __new__(
                 serializable_error_info, "serializable_error_info", SerializableErrorInfo
             ),
         )
+
+
+@whitelist_for_serdes
+class ResourceVerificationRequest(
+    NamedTuple(
+        "_ResourceVerificationRequest",
+        [
+            ("repository_origin", ExternalRepositoryOrigin),
+            ("instance_ref", Optional[InstanceRef]),
+            ("resource_name", str),
+        ],
+    )
+):
+    def __new__(
+        cls,
+        repository_origin: ExternalRepositoryOrigin,
+        instance_ref: Optional[InstanceRef],
+        resource_name: str,
+    ):
+        return super(ResourceVerificationRequest, cls).__new__(
+            cls,
+            repository_origin=check.inst_param(
+                repository_origin, "repository_origin", ExternalRepositoryOrigin
+            ),
+            instance_ref=check.opt_inst_param(instance_ref, "instance_ref", InstanceRef),
+            resource_name=check.str_param(resource_name, "resource_name"),
+        )
+
+
+@whitelist_for_serdes
+class ResourceVerificationResult(
+    NamedTuple(
+        "_ResourceVerificationResult",
+        [
+            ("response", VerificationResult),
+            ("serializable_error_info", Optional[SerializableErrorInfo]),
+        ],
+    )
+):
+    def __new__(
+        cls, response: VerificationResult, serializable_error_info: Optional[SerializableErrorInfo]
+    ):
+        return super(ResourceVerificationResult, cls).__new__(
+            cls,
+            response=check.inst_param(response, "response", VerificationResult),
+            serializable_error_info=check.opt_inst_param(
+                serializable_error_info, "serializable_error_info", SerializableErrorInfo
+            ),
+        )
+
+
+@whitelist_for_serdes
+class UserCodeExecutionType(Enum):
+    """Represents the type of user code execution request.
+
+    This is used to determine and verify specific request and result types when unpacking the generic
+    UserCodeExecutionRequest and UserCodeExecutionResult types.
+    """
+
+    RESOURCE_VERIFICATION = "resource_verification"
```

Comment:
I am not confident that our enum forward compat story is great - so we may want to use a string or GRPC level enum